### PR TITLE
[SLAM] Remove the deprecated implementation

### PR DESCRIPTION
**Pull request by DonnaWuDongxia**
- remove files under worker folder
- remove event flags related code
- rename slam_module and slam_runner
- updated related test cases

### CODE REVIEW

**Comment by DonnaWuDongxia** [comment]
@kenny-y @haoyunfeix @lisa0314 please help to review.

File: src/slam/instance.cpp [hunk 1]
```diff
@@ -5,64 +5,44 @@
 #include "instance.h"
 
 Instance::Instance() {
-  runner_ = std::make_shared<SlamRunner>();
 }
 
 Instance& Instance::operator=(const Instance& rhs) {
   if (&rhs != this) {
-    this->runner_ = rhs.runner();
+    this->js_this_.Reset(Nan::New(rhs.js_this_));
```
Comment (kenny-y) [inline src/slam/instance.cpp:12]: Do we need this `operator =` ? Has it ever been invoked in runtime?

**Review by kenny-y — APPROVED** [review]
Didn't find anything wrong in my perspective. Let's see how it behaves in runtime.

**Comment by halton** [comment]
Merge it because @kenny-y approve it.

**Pull request merged**